src/shaders/_prelude_shadow.fragment.glsl

```diff
@@ -56,25 +56,25 @@ highp float shadow_occlusion_0(highp vec4 pos, highp float bias) {
     highp vec2 uv23 = uv03 + vec2(2.0 * s, 0);
     highp vec2 uv33 = uv03 + vec2(3.0 * s, 0);
 
-    highp float o00 = shadow_sample_0(uv00, fragDepth);
-    highp float o10 = shadow_sample_0(uv10, fragDepth);
-    highp float o20 = shadow_sample_0(uv20, fragDepth);
-    highp float o30 = shadow_sample_0(uv30, fragDepth);
+    highp float o00 = shadow_sample_0(uv00, compare0);
+    highp float o10 = shadow_sample_0(uv10, compare0);
+    highp float o20 = shadow_sample_0(uv20, compare0);
+    highp float o30 = shadow_sample_0(uv30, compare0);
 
-    highp float o01 = shadow_sample_0(uv01, fragDepth);
-    highp float o11 = shadow_sample_0(uv11, fragDepth);
-    highp float o21 = shadow_sample_0(uv21, fragDepth);
-    highp float o31 = shadow_sample_0(uv31, fragDepth);
+    highp float o01 = shadow_sample_0(uv01, compare0);
+    highp float o11 = shadow_sample_0(uv11, compare0);
+    highp float o21 = shadow_sample_0(uv21, compare0);
+    highp float o31 = shadow_sample_0(uv31, compare0);
 
-    highp float o02 = shadow_sample_0(uv02, fragDepth);
-    highp float o12 = shadow_sample_0(uv12, fragDepth);
-    highp float o22 = shadow_sample_0(uv22, fragDepth);
-    highp float o32 = shadow_sample_0(uv32, fragDepth);
+    highp float o02 = shadow_sample_0(uv02, compare0);
+    highp float o12 = shadow_sample_0(uv12, compare0);
+    highp float o22 = shadow_sample_0(uv22, compare0);
+    highp float o32 = shadow_sample_0(uv32, compare0);
 
-    highp float o03 = shadow_sample_0(uv03, fragDepth);
-    highp float o13 = shadow_sample_0(uv13, fragDepth);
-    highp float o23 = shadow_sample_0(uv23, fragDepth);
-    highp float o33 = shadow_sample_0(uv33, fragDepth);
+    highp float o03 = shadow_sample_0(uv03, compare0);
+    highp float o13 = shadow_sample_0(uv13, compare0);
+    highp float o23 = shadow_sample_0(uv23, compare0);
+    highp float o33 = shadow_sample_0(uv33, compare0);
```

Comment:
Changes up to this point were done to make a statement unique for the the two cascades. This was needed because of the metal shader generator.